### ISSUE TITLE
False Positive: "Use of moved value"

### ISSUE THREAD

**Issue by adamiao** [intellij-rust/intellij-rust#4802]
## Environment

* **IntelliJ Rust plugin version:** 0.2.112.2148-193
* **Rust toolchain version:** rustc 1.40.0 and rustup 1.21.1
* **IDE name and version:** PyCharm 2019.3.1 (Community Edition)
* **Operating system:** Linux Mint 19.3

## Problem description
I get **"Use of moved value"** errors (not warnings, but the actual red exclamation mark) from the IDE for the variables `index_a` and `index_b` within the `swap` function for the code below.

```rust
fn swap<T>(index_a: usize, index_b: usize, arr: &mut [T])
    where T: Copy {
    let temp = arr[index_a];
    arr[index_a] = arr[index_b];
    arr[index_b] = temp;
}

fn main() {
    let mut v = [0, 1, 2, 3];
    println!("{:?}", v); //[0, 1, 2, 3]
    swap(1, 2, &mut v);
    println!("{:?}", v); //[0, 2, 1, 3]
}
```

The code compiles without any errors in both the IDE as well as in the _Rust Playground_ (https://play.rust-lang.org/)

## Steps to reproduce

To replicate the issue please use the same code in the IDE.

**Comment by Undin** [comment]
@adamiao Thanks for the report!
Looks like `Copy` trait cannot be found in stdlib and plugin doesn't know that `usize` implements `Copy`.
I can reproduce it only when you don't have any cargo project or stdlib cannot be found.
Could you check that you have al least one cargo project (see Cargo tool window) and the plugin knows where Rust stdlib is located (see Rust settings)?

![image](https://user-images.githubusercontent.com/2539310/71564510-447fe880-2ab2-11ea-8377-e6a8e9f9fe28.png)


![image](https://user-images.githubusercontent.com/2539310/71564501-2c0fce00-2ab2-11ea-82cc-80db0662e911.png)

**Comment by adamiao** [comment]
@Undin You're absolutely right! Now that I have a new cargo project just for this code **I do not observe the reported error**.

I'm very sorry for any extra work this may have caused.
